### PR TITLE
single qubit circuits should not 'coupling_map = None'

### LINKED ISSUE #2038: Cannot map a single qubit to a given device qubit that is not zero
### Information

- **Qiskit Terra version**: master
- **Python version**:
- **Operating system**:

### What is the current behavior?
It is not possible to map a single qubit circuit to a device qubit that is not zero.

```python

qr = QuantumRegister(1, 'qr')
qc = QuantumCircuit(qr)
qc.h(qr[0])

layout = Layout({(qr,0): 19})
backend = IBMQ.get_backend('ibmq_16_melbourne')
new_circ = transpile(qc, backend, initial_layout=layout)
new_circ.draw()
```

Screen Shot 2019-03-28 at 11 18 08 AM

Note also that the layout used is not valid for the device, but no error is thrown.  I think all of this is due to this:

https://github.com/Qiskit/qiskit-terra/blob/ef46e442e4320500c847da617aadd5476bca4b70/qiskit/transpiler/transpiler.py#L190

that sets the coupling map to `None`, and therefore the transpiler skips the swap mapper that currently implements the mapping to device qubits.  Increasing the quantum register size to two or more gives a circuit with correct mapping, and which is the width of the device.

### PR DESCRIPTION
**Pull request by 1ucian0**
Fixes #2038 

In #2038, @nonhermitian notices that single qubit have no restrictions in the coupling map and, therefore, no restrictions in the layout definition, because:

https://github.com/Qiskit/qiskit-terra/blob/ef46e442e4320500c847da617aadd5476bca4b70/qiskit/transpiler/transpiler.py#L190

I could trace this "exceptional" behavior to  https://github.com/Qiskit/qiskit-terra/commit/5c736a6d94e7d97446f3f93da88a7f5cde157c87#diff-d84cfcad92182447556156d4d8a14c69R77, where @jaygambetta leaves a comment saying:

> `#THIS IS A BUG IF the second circuit has more than 1 qubit`

I have no idea what that means. Maybe @jaygambetta can comment about it?

This PR removes the exceptional behavior and adds a regression test for #2038.

### CODE REVIEW

**Comment by 1ucian0** [comment]
An even older reference to the removed branch https://github.com/Qiskit/qiskit-terra/pull/705/files#diff-d84cfcad92182447556156d4d8a14c69L133 (comes from an `unknown repository`)

**Comment by ajavadia** [comment]
I think the original issue with #2038 is that the laying `q[0]` onto any physical qubit other than `0` fails. Raising an error for the case where physical qubit is out of range is a secondary issue.

The test here covers the second case, but not the first. Can you also make a test to make sure mapping the single virtual qubit to some other physical qubit (say 12), works?

**Comment by 1ucian0** [comment]
I added https://github.com/Qiskit/qiskit-terra/pull/2110/files#diff-f25d0847ceb8a575a84a3a4c65be8709R195 Is that what you mean?

File: test/python/compiler/test_compiler.py [hunk 1]
```diff
@@ -171,6 +171,48 @@ def test_transpiler_layout_from_intlist(self):
...
+                [5, 4], [5, 6], [5, 9], [6, 8], [7, 8],
+                [9, 8], [9, 10], [11, 3], [11, 10],
+                [11, 12], [12, 2], [13, 1], [13, 12]]
+
+        result = transpile(qc, backend=None, coupling_map=cmap, basis_gates=['u2'],
+                           initial_layout=layout)
+
+        qr = QuantumRegister(14, 'q')
+        expected = QuantumCircuit(qr)
+        expected.u2(0, 3.1416, qr[12])
```
Comment (ajavadia) [inline test/python/compiler/test_compiler.py]: can you actually go to Qobj here (use `qiskit.compiler.assemble_circuits`), and check that the QobjInstruction  for`u2` is applied to physical qubit 12? That will be the most reassuring means by which we can say layout was done correctly.

Comment (1ucian0) [reply test/python/compiler/test_compiler.py]: done in f6db1d28

**Pull request merged**